openreview/conference/invitation.py

```diff
@@ -600,27 +605,144 @@ def set_recommendation_invitation(self, conference, start_date, due_date, notes_
+                id = conference.get_recommendation_id(number = note.number),
+                cdate = tools.datetime_millis(start_date),
+                duedate = tools.datetime_millis(due_date),
+                expdate = tools.datetime_millis(due_date + datetime.timedelta(minutes = SHORT_BUFFER_MIN)),
                 super = recommendation_invitation.id,
                 invitees = [conference.get_program_chairs_id(), conference.get_id() + '/Paper{number}/Area_Chairs'.format(number = note.number)],
                 writers = [conference.get_id()],
                 signatures = [conference.get_id()],
                 multiReply = True,
-                reply = {
-                    'forum': note.id,
-                    'content': {
-                        'tag': {
-                            'description': 'Recommend reviewer',
-                            'order': 1,
-                            'required': True,
-                            'values-dropdown': reviewers
-                        }
-                    }
-                }
+                reply = reply
             )
             paper_recommendation_invitation = self.client.post_invitation(paper_recommendation_invitation)
-            print('Posted', paper_recommendation_invitation.id)
+
+
+    def set_registration_invitation(self, conference, start_date, due_date, with_area_chairs):
+
+        invitees = []
+        if with_area_chairs:
+            invitees.append(conference.get_area_chairs_id())
+        invitees.append(conference.get_reviewers_id())
+
+        subj_desc = 'To properly assign papers to reviewers, we ask that reviewers provide their areas of expertise from among the provided list of subject areas. Please submit your areas of expertise by selecting the appropriate options from the "Subject Areas" list.\n\n'
+
+        coi_desc = 'In order to avoid conflicts of interest in reviewing, we ask that all reviewers take a moment to update their OpenReview profiles with their latest information regarding work history and professional relationships. After you have updated your profile, please confirm that your OpenReview profile is up-to-date by selecting yes in the "Profile Confirmed" section.\n\n'
+
+        tpms_desc = 'In addition to subject areas, we will be using the Toronto Paper Matching System (TPMS) to compute paper-reviewer affinity scores. Please take a moment to sign up for TPMS and/or update your TPMS account with your latest papers. Then, please ensure that the email address that is affiliated with your TPMS account is linked to your OpenReview profile. After you have done this, please confirm that your TPMS account is up-to-date by selecting yes in the "TPMS Account Confirmed" section.\n\n'
+
+
+        # Create super invitation with a webfield
+        registration_parent_invitation = openreview.Invitation(
+            id = conference.get_id() + '/-/Parent/Registration',
```

Comment:
I think this is just a matter of personal taste, but what do you think about naming this "Registration/Super" instead? or "Super/Registration"?

No strong argument about it, I think it mostly doesn't matter. But "Parent" doesn't feel descriptive enough to me.

Reply:
Any name is fine for me. It is the actual UAI configuration. 

I think we shouldn't have to create a fake forum to submit the reviewer registration.

Reply:
Yeah, I agree, ideally we wouldn't do that

Reply:
Ok, I renamed the invitation.